### DISCUSSION TITLE
how use `useLogger` ?

### DISCUSSION THREAD

**Discussion by zss1004504439** [nuxt/nuxt#22151] [Questions] [answered]
## code
```js
// pages/index.vue

import { useLogger } from '@nuxt/kit'
const logger = useLogger('useHttp')
logger.info('logger content ...')
```
## result 
![image](https://github.com/nuxt/nuxt/assets/24915880/2b983d33-1361-4f37-a676-1f848c4cb940)

**Comment by Jannchie** [comment]
`useLogger` appears to be a composition that is intended for use in a Node.js environment.  It may not be suitable for use in .vue files.

If you must use it, you may need to determine that `typeof window === 'undefined'`.

**Reply by gustavoconter** [reply, marked as answer]
Just adding more context. `@nuxt/kit` is meant to be used while developing nuxt modules and not meant to be imported during runtime (components, vue composables, pages, plugins, or server routes)

[Here is the short docs for Nuxt Kit](https://nuxt.com/docs/guide/going-further/kit)

**Reply by maicss** [reply]
> Just adding more context. `@nuxt/kit` is meant to be used while developing nuxt modules and not meant to be imported during runtime (components, vue composables, pages, plugins, or server routes)
> 
> [Here is the short docs for Nuxt Kit](https://nuxt.com/docs/guide/going-further/kit)

<img width="705" alt="image" src="https://github.com/nuxt/nuxt/assets/9784767/ce95f76b-f22b-4a15-b4a6-983362001445">

So this is expected behavior? And only for someone who want develop a nuxt module? Whether add `this only for who develop a nuxt module, do not use this in your project` to the document?

And as it use `consola`, why not use it in server/browser side? `consola` document says it support both server and browser.

**Reply by TheAlexLichter** [reply]
Nuxt Kit should not be used during runtime. 

Instead, just use consola for the logger 👍

**Reply by s0h311** [reply]
Consider checking out Pino: https://github.com/pinojs/pino. It's super easy to integrate to your project and start using it.

**Reply by maicss** [reply]
OK, if there any chance not expose `@nuxt/kit` to user develop environment, only who mark(like a field in nuxt.config.ts) they want to build a nuxt module? Thanks.
